tool/gravity/cli/ops.go

```diff
@@ -87,110 +88,155 @@ func appPackage(env *localenv.LocalEnvironment) error {
 }
 
 func uploadUpdate(env *localenv.LocalEnvironment, opsURL string) error {
-	// create local environment with gravity state dir because the environment
-	// provided above has upgrade tarball as a state dir
-	localStateDir, err := localenv.LocalGravityDir()
+	clusterOperator, err := env.SiteOperator()
```

Comment:
If `env` has the tarball as its state dir (as the comment that you removed above said), then this will attempt to look for "ops" credentials in the tarball state (and won't find them), no?

Reply:
`env` is not (and should not be) the tarball directory - it is whatever state directory has been specified with `--state-dir` (and would be the local gravity state directory if unspecified).
I'm adding an explicit (but optional) `--data-dir` to the command to specify the tarball directory and avoid confusion with the state directory.